aptos-move/framework/aptos-framework/sources/configs/gas_schedule.move

```diff
@@ -23,16 +31,27 @@ module aptos_framework::gas_schedule {
         entries: vector<GasEntry>
     }
 
+    /// Called during a module upgrade.
+    /// Creates and publishes the resources that are added at a later time.
+    public fun upgrade_module(aptos_framework: &signer) {
+        system_addresses::assert_aptos_framework(aptos_framework);
+
+        if (!exists<GasFeatureVersion>(signer::address_of(aptos_framework))) {
+            move_to<GasFeatureVersion>(aptos_framework, GasFeatureVersion { major: GENESIS_GAS_FEATURE_VERSION });
+        }
+    }
+
```

Comment:
@movekevin it would be really great if you could offer me some feedback on the public interfaces, especially the way I handle the module upgrade (publishing of the new resource)